### ISSUE TITLE
image is not recognized

### ISSUE THREAD

**Issue by msth** [Wh1terat/aQRootG3#5]
Hello, 
thanks for the tool unfortunately the generated qr code is not recognized by the camera. I tried it with the FW version 3.3.2_0003.0004 and 3.3.4_0007.0004.

**Comment by Wh1terat** [comment]
One or two others have faced this problem which leads me to believe it's environmental.

It's a lot of data to fit into a qrcode and error correction is set to it's highest to avoid any corruption during scanning.

A mouse cursor over the qrcode or reflection on a screen is enough to prevent this, during my testing (largely at night) I even had to place my finger over the led on the front of the camera to prevent the light reflecting from the screen whilst scanning.

Ideal distance is approx. 15cm from the camera for scanning.

**Comment by msth** [comment]
Thank you for your prompt reply. I copied the picture to my smartphone. And from there tried different positions. also locked the ir led. Unfortunately without success.
when I scan the code with my  phone (Pixel 6), "unknown encoding" is displayed.

**Comment by Wh1terat** [comment]
I wonder if it's something to do with the image output.

Although a bit inconvenient, can you try scanning the code printed to terminal when you run the script instead?

**Comment by ninamarie8253** [comment]
Honestly I’m beginning to think it’s something to do with the USA Aqara camera Vs the China version. You used the china version which works, I'm using the US version, which didn’t work no matter which device or lighting I tried to show the QR code on. Yet it picked up on the Aqara QR code immediately, so it couldn’t be environmental in that case.

**Comment by msth** [comment]
@Wh1terat it is absolutely the same picture. I tryed it anyway, and unfortunately without success

@ninamarie8253 where can i see if it is the china or us version? it is at least "made in china"  ;-)

**Comment by Wh1terat** [comment]
Can you try v0.1 ?

Payload is much smaller, might be worth a try.

https://github.com/Wh1terat/aQRootG3/blob/8f7b29662d170f508d56db06b2c4362e9a694b8d/aQRootG3.py

**Comment by ninamarie8253** [comment]
I think in general if you bought it from aliexpress, its the china version. if you bought it say off amazon in the US, its the US version.  If your device responds in chinese when you reset it, that probably means its the chinese version. Mine responds in english. This is all a guess btw.
    On Monday, January 31, 2022, 09:30:09 AM EST, msth ***@***.***> wrote:  
 
 


@Wh1terat it is absolutely the same picture. I tryed it anyway, and unfortunately without success

@ninamarie8253 where can i see if it is the china or us version? it is at least "made in china" ;-)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by msth** [comment]
OK, then i've got an u.s. Version

**Comment by Wh1terat** [comment]
The hardware and even the firmware is the same, it's just a bunch of properties that define the region and default language. I'd be amazed if it's something to do with it being a US model - but also I can't ignore the fact the only 2 users I know to have problems both have US models.

@rezmus - anything you're aware of?

**Comment by rezmus** [comment]
i would first check qr code. generate new one with ssid/pass = test/test and parse image with online qr decoder

https://zxing.org/w/decode.jspx

check if parsed result match this one

```
b=#!/bin/sh\nfw_manager.sh -r\npasswd -d $USER\nfw_manager.sh -t -k&d=a;y=/data/scripts/post_init.sh;"fw_man"ager.sh -t -f;echo -e `agetprop persist.app.bind_key`>$y;tail -n2 $y|sh&x=.=/.&y=.=/.&l=en
```

if it is 100% the same, generate qr code with your wifi ssid/pass. then do the same in aqara app. make screenshot of this qr. parse both qr codes with online decoder. compare x= and y= if they are the same on both qr codes.

**Comment by msth** [comment]
Hello @rezmus , 
thanks for the Link.

Unfortunately I get the message that the picture is too big

![Unbenannt](https://user-images.githubusercontent.com/1927680/151917870-bd07b9d6-0e8e-409b-8c32-5c57a1504b31.png)


attached my qr code with test as ssid and psk.

![test](https://user-images.githubusercontent.com/1927680/151917599-136ad129-283e-487b-9577-58257549e154.svg)

**Comment by Wh1terat** [comment]
[CyberChef](https://gchq.github.io/CyberChef/#recipe=Parse_QR_Code(true)) works well for this.

Decodes fine for me so not an issue with segno library.

**Comment by rezmus** [comment]
is there any camera reaction on custom qr? like led or sound? if not, try to generate qr with original data (same as app does) and check if it is recognized. you can also print qr on paper.

**Comment by msth** [comment]
> [CyberChef](https://gchq.github.io/CyberChef/#recipe=Parse_QR_Code(true)) works well for this.
> 
> Decodes fine for me so looking less likely to be an issue with segno library.
> 
> Decoded and re-encoded using cyberchef, can you see if the camera even tries to read that?
> 
> ![image](https://user-images.githubusercontent.com/1288525/151954972-3a93cab6-dc65-4387-9623-96965f5ecceb.png)

with your image i receive "wifi connecting. please wait". 

which segno library is used? tried it with 1.3.3-1 and 1.4.1-1

**Comment by msth** [comment]
I found the problem. It's a special character.
I have a ! in PSK

**Comment by msth** [comment]
i changed the psk for a short time. now it's working. Thanks for your help

**Comment by Wh1terat** [comment]
How bizarre! 
Thank you @msth  I'll do some testing and try to get a fix out shortly.

**Comment by Wh1terat** [comment]
@msth 
Fix in dev branch
https://github.com/Wh1terat/aQRootG3/tree/dev

_Should_ work with pretty much anything now.

```
x@x:[~/src/aQRootG3]$ ./aQRootG3.py 😅 тестовый

b=#!/bin/sh\nfw_manager.sh -r\npasswd -d $USER\nfw_manager.sh -t -k&d=a;y=/data/scripts/post_init.sh;"fw_man"ager.sh -t -f;echo -e `agetprop persist.app.bind_key`>$y;tail -n2 $y|sh&x="C!C!<!)&y="$!&!t!Y"$!%"$!&!t!b!t!V"$!/!t!]&l=en 
``` 

@rezmus 
FYI ^

**Comment by Wh1terat** [comment]
More testing done and happy now.

Merged into main branch @ 17678ed796ae7cad6053d83f353c4e29551b2b18
